lib/spack/spack/binary_distribution.py

```diff
@@ -712,6 +714,21 @@ def get_buildfile_manifest(spec):
     root = spec.prefix
     visit_directory_tree(root, visitor)
 
+    # Collect a list of prefixes for this package and it's dependencies, Spack will
+    # look for them to decide if text file needs to be relocated or not
+    prefixes = [d.prefix for d in spec.traverse(root=False, deptype="all") if not d.external]
+    prefixes.append(spec.prefix)
+
+    # Adopted from relocate.unsafe_relocate_text
+    # Create a list regexes matching collected prefixes
+    compiled_prefixes = []
+
+    for prefix in prefixes:
+        prefix_bytes = prefix.encode("utf-8")
+        prefix_rexp = re.compile(b"(?<![\\w\\-_/])([\\w\\-_]*?)%s([\\w\\-_/]*)" % prefix_bytes)
```

Comment:
I see you copied this directly from `relocate_text`, which is good, except that that code is broken. On Linux paths are a null-terminated sequence of bytes. So, this must obviously be escaped to ensure `prefix_bytes` is a literal binary string. Can you:

1. create a function `utf8_path_to_binary_regex(path)` that takes a string and outputs a compiled regex
2. use `map(utf8_path_to_binary_regex, prefixes)` in both locations
3. add a unit test where you check that `utf8_path_to_binary_regex("[a-z]")` or so works on the string `b"[a-z]"` but not on `b"abc"`?

Reply:
@haampie sorry, I don't fully understand what you are saying here (mostly about zero-termination). 

As far as I understand, I need to refactor the borrowed code into `relocate.utf8_path_to_binary_regex` while also sanitizing `prefix` argument. I don't think we need a unit test for that, since there is a handy [`re.escape`](https://docs.python.org/3/library/re.html#re.escape) function provided by Python itself.

Reply:
Yeah, just add

```python
import re
def utf8_path_to_binary_regex(path):
    """Create a (binary) regex that matches the input path in utf8""" 
    return re.compile(b"(?<![\\w\\-_/])([\\w\\-_]*?)%s([\\w\\-_/]*)" % re.escape(path.encode("utf-8")))
```

and add a test

```python
import re
def test_text_relocation_regex():
    assert utf8_path_to_binary_regex("/[a-z]/").search(b"This does not match /a/, but this does: /[a-z]/.").group(0) == b"/[a-z]/"
True
```

Reply:
And then use that function in both locations where the regex is used.